zerver/views/realm.py

```diff
@@ -152,7 +152,17 @@ def update_realm(
             message_retention_days_raw, Realm.MESSAGE_RETENTION_SPECIAL_VALUES_MAP
         )
 
-    if invite_to_realm_policy is not None and not user_profile.is_realm_owner:
+    if (
+        invite_to_realm_policy is not None or invite_required is not None
+    ) and not user_profile.is_realm_owner:
+        raise OrganizationOwnerRequired()
+
+    if (
+        emails_restricted_to_domains is not None or disallow_disposable_email_addresses is not None
+    ) and not user_profile.is_realm_owner:
+        raise OrganizationOwnerRequired()
+
+    if waiting_period_threshold is not None and not user_profile.is_realm_owner:
```

Comment:
I have grouped these settings according to their use, or do we need a separate if statement for each one of them.

Reply:
It might be slightly better to have individual lines to help ensure we have test coverage for both cases, but this is probably fine.